src/addrs/addrs.go

```diff
@@ -95,6 +139,56 @@ func (a *Addrs) NewAddress() (string, error) {
 	return chosenAddr, nil
 }
 
+func (a *Addrs) GetAll() ([]string, error) {
+	a.RLock()
+	defer a.RUnlock()
+
+	return a.all.GetAll()
+}
+
+func (a *Addrs) GetUsed() ([]string, error) {
+	a.RLock()
+	defer a.RUnlock()
+
+	return a.used.GetAll()
+}
+
+func (a *Addrs) SetUsed(addr string) error {
+	a.RLock()
+	defer a.RUnlock()
+
+	err := a.used.Put(addr)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *Addrs) GetFree() ([]string, error) {
+	a.RLock()
```

Comment:
why? We have not bucket with free address, just check if all aren't exist in used

Reply:
because otherwise has race condition. but, this method will be very slow, it loads all of the keys, then it queries in a loop.  I am thinking Store should manage "used" and "all" buckets